packages/kbn-es-archiver/src/lib/docs/index_doc_records_stream.ts

```diff
@@ -32,6 +32,7 @@ export function createIndexDocRecordsStream(
     await client.helpers.bulk(
       {
         retries: 5,
+        refreshOnCompletion: true,
```

Comment:
@elastic/kibana-operations we were not ensuring indices were refreshed after injecting docs, leading to yet another source of flakiness. E.g.:
https://buildkite.com/elastic/kibana-pull-request/builds/136166#0188d3d0-6c07-4b95-9ab5-248be760ae68

Reply:
Good catch - thank you.

Reply:
It turns out this change has uncovered some poorly designed tests that were working by the mere fact that ES indices were not refreshed after `esArchiver.load`.

I'm looking at the [build failures](https://buildkite.com/elastic/kibana-pull-request/builds/136855#_), and I confirm the first one falls in that category. I might use a different PR to address these failures, or this one will end up involving a lot of teams.

Reply:
thank you, I had an expectation it is already doing refresh.

Reply:
This is getting out of hand for the PR at stakes, so I created a separate PR to tackle this:

https://github.com/elastic/kibana/pull/160428

I reverted changes in this one. Let's see how CI likes it.